hydroflow_plus_example_flow/src/lib.rs

```diff
@@ -0,0 +1,29 @@
+use hydroflow_plus::{q, quse, HydroflowContext, HydroflowNode, RuntimeData};
+
+fn filter_by_regex<'a, S: AsRef<str> + 'a>(
+    input: HydroflowNode<'a, String>,
+    pattern: RuntimeData<S>,
+) -> HydroflowNode<'a, String> {
+    quse!(::regex::Regex);
+
+    input.filter(q!({
+        let regex = Regex::new(pattern.as_ref()).unwrap();
+        move |x| regex.is_match(x)
+    }))
+}
+
+#[hydroflow_plus::flow(String)]
```

Comment:
We have to provide a canonical type for each type parameter, because the never type doesn't (yet) implement all traits.